download_data.py

```diff
@@ -3,32 +3,71 @@
 import time
 import sys
 import os
+import subprocess
+import glob
 
 from bitmind.constants import DATASET_META
 
+def clear_cache(cache_dir):
+    """Clears lock files and incomplete downloads from the cache directory."""
+    # Find lock and incomplete files
+    lock_files = glob.glob(cache_dir + "/*lock")
+    incomplete_files = glob.glob(cache_dir + "/downloads/**/*.incomplete", recursive=True)
+    try:
+        if lock_files:
+            subprocess.run(["rm", *lock_files], check=True)
```

Comment:
Interesting reads about the dangers of enabling shell=True for subprocess commands. The default is shell=False, which is recommended.

https://www.codiga.io/blog/python-subprocess-security/
https://stackoverflow.com/questions/3172470/actual-meaning-of-shell-true-in-subprocess